myriad-scheduler/src/main/java/com/ebay/myriad/Main.java

```diff
@@ -190,6 +196,27 @@ private void startNMInstances(Injector injector) {
       }
     }
 
+    /**
+     * Create NMProfile for any configured service
+     * @param injector
+     */
+    private void initJavaBasedTaskConfiguration(Injector injector) {
+      LOGGER.info("Initializing JavaBasedTaskConfiguration");
+      NMProfileManager profileManager = injector.getInstance(NMProfileManager.class);
+
+      Map<String, AuxTaskConfiguration> auxServicesConfigs = 
+          injector.getInstance(MyriadConfiguration.class).getAuxTaskConfigurations();
+      if (auxServicesConfigs != null) {
+        for (Map.Entry<String, AuxTaskConfiguration> entry : auxServicesConfigs.entrySet()) {
+          AuxTaskConfiguration config = entry.getValue();
+          final Double cpu = config.getCpus().or(AuxTaskConfiguration.DEFAULT_CPU);
+          final Double mem = config.getJvmMaxMemoryMB().or(AuxTaskConfiguration.DEFAULT_MEMORY);
+          
+          profileManager.add(new NMProfile(entry.getKey(), cpu, mem));
```

Comment:
I propose to rethink the meaning of NMProfile - NMProfile is really a set of resources needed to run some tasks: those tasks could be containers, or other types of tasks, that is why I reused NMProfile. We could name it differently though. I am not sure it is a good idea to come up with yet another abstraction for service resources that will pretty much will repeat NMProfile structure. Let's discuss though